backend/main.go

```diff
@@ -1,22 +1,18 @@
 package main
 
 import (
-	"log"
-	"net/http"
+	"flag"
 
-	"github.com/gorilla/mux"
-	"github.com/ibeckermayer/teleport-interview/backend/internal/handlers"
+	"github.com/ibeckermayer/teleport-interview/backend/internal/server"
 )
 
 func main() {
-	router := mux.NewRouter()
+	port := flag.Int("port", 8000, "Port to serve the app from")
+	certFilePath := flag.String("cert", "../certs/localhost.crt", "Relative path to a valid SSL cert")
+	keyFilePath := flag.String("key", "../certs/localhost.key", "Relative path to the cert's private key")
 
-	router.Handle("/api/login", http.HandlerFunc(handlers.Login)).Methods("POST")
+	cfg := server.NewConfig(*port, *certFilePath, *keyFilePath)
+	srv := server.New(cfg)
```

Comment:
This is not an issue and you don't need to make this change, but a nice pattern we use is the following:

```go
srv, err := server.New(&server.Config{
  Port:        *port,
  Certificate: *certFilePath,
  Key:         *keyFilePath,
})
```

Reply:
Roger that, will keep that in my back pocket assuming the rest of the interview process goes well :)